src/charm.py

```diff
@@ -555,28 +559,28 @@ def _wp_is_installed(self):
         logger.debug("Check if WordPress is installed")
         return self._run_wp_cli(["wp", "core", "is-installed"]).return_code == 0
 
-    def _parse_database_endpoints(self, endpoint: Optional[str]) -> Optional[str]:
+    def _parse_database_endpoints(
+        self, endpoint: Optional[str]
+    ) -> Tuple[Optional[str], Optional[int]]:
         """Retrieve a single database endpoint.
 
         Args:
             endpoint: An endpoint of format host:port
 
         Returns:
-            Hostname of database running on port 3306. None if no endpoints are provided.
-            Note that WordPress will throw MySQL Error when supplying a port with the hostname
-            (i.e. host:port).
+            Hostname and port of database. None if no endpoints are provided.
 
         Raises:
-            WordPressBlockedStatusException: Provided endpoint contains port other than 3306.
+            RuntimeError: Provided endpoint contains port other than 3306.
         """
         if not endpoint:
-            return None
+            return None, None
         host_port = endpoint.split(":")
-        if len(host_port) == 2 and host_port[1] != "3306":
-            raise exceptions.WordPressBlockedStatusException(f"Invalid port {host_port[1]}")
-        # The endpoint might not contain port, we assume it to be 3306. If not, it will be caught
-        # by `_test_database_connectivity` function later on.
-        return host_port[0]
+        if len(host_port) == 1:
+            return host_port[0], 3306
```

Comment:
Bit picky but, can you define the port in a constant?

Reply:
Updated, thanks!